consensus/state.go

```diff
@@ -560,7 +581,7 @@ func (cs *ConsensusState) EnterPropose(height int, round int) {
 		log.Info("EnterPropose: Not our turn to propose", "proposer", cs.Validators.Proposer().Address, "privValidator", cs.privValidator)
 	} else {
 		log.Info("EnterPropose: Our turn to propose", "proposer", cs.Validators.Proposer().Address, "privValidator", cs.privValidator)
-		cs.decideProposal(height, round)
+		cs.decideProposal(height, round) // if this takes longer than timeoutPropose we'll catch it in a later EnterPropose
```

Comment:
Kinda confusing comment.  For one, just because this times out doesn't mean that we will definitely EnterPropose for the next round, depending on network state.

"if this takes longer than timeoutPropose we'll EnterPrevote() and prevote nil or whatever we're locked on".